src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+
+        FigureSupplier figureSupplier = new FigureSupplier();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        int middle = figures.length / 2;
+        for (int i = 0; i < middle; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+            figures[i].setColor(colorSupplier.getRandomColor());

```

Comment:
figureSupplier should supply us Figure with RandomColor. Why do we need to do this extra work here?